docs/verify_cstor_bds.sh

```diff
@@ -0,0 +1,74 @@
+set -e
+ns=$1
+if [[ $ns == "" ]]; then
+    ns="openebs"
+fi 
+
+find_bd_for_devlink() {
+        devlink=$1
+        bdList=$(kubectl -n $ns get blockdevices -o jsonpath='{.items[*].metadata.name}')
+        for bd in $bdList
+        do
+                links=$(kubectl -n $ns get blockdevices $bd -o jsonpath="{.spec.devlinks[*].links}")
+                # remove the list [] from by-id/by-path links output
+                links=$(echo $links | tr -d [ | tr -d ])
+                if [[ $links == "" ]]; then
+                        links=$(kubectl -n $ns get blockdevices $bd -o jsonpath="{.spec.path}")
+                fi
+                for link in $links
+                do
+                        if [[ "$devlink" == *"$link"* ]]; then
+                                state=$(kubectl -n $ns get blockdevices $bd -o jsonpath="{.status.state}")
+                                claimState=$(kubectl -n $ns get blockdevices $bd -o jsonpath="{.status.claimState}")
+                                if [[ $state == "Active" && $claimState == "Unclaimed" ]]; then
+                                        echo "$bd"
+                                        break
+                                fi
+                        fi
+                done
+        done
+}
+
+cspList=$(kubectl get csp -o jsonpath='{.items[*].metadata.name}')
+for csp in $cspList
+do
+        echo "Verifying blockdevices on $csp"
+        pod=$(kubectl -n $ns get pods -l openebs.io/cstor-pool=$csp -o jsonpath="{.items[?(@.status.phase=='Running')].metadata.name}")
+        devlinks=$(kubectl -n $ns exec -it $pod -c cstor-pool -- zpool status -P | grep \/dev | awk '{print $1}')
```

Comment:
Q: If there were no pods in running status this step should fail.

Reply:
Added the check